spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/security/IapAuthenticationAutoConfiguration.java

```diff
@@ -109,33 +106,24 @@ public AudienceProvider appEngineBasedAudienceProvider(GcpProjectIdProvider proj
 
 	@Bean
 	@ConditionalOnMissingBean
-	@ConditionalOnGcpEnvironment({GcpEnvironment.COMPUTE_ENGINE, GcpEnvironment.KUBERNETES_ENGINE})
-	public AudienceProvider computeEngineBasedAudienceProvider(GcpProjectIdProvider projectIdProvider) {
-		return new ComputeEngineAudienceProvider(projectIdProvider);
-	}
-
-	@Bean
-	@ConditionalOnBean(AudienceProvider.class)
-	@ConditionalOnMissingBean
-	public AudienceValidator environmentBasedAudienceValidator(AudienceProvider audienceProvider) {
+	public AudienceValidator audienceValidator(AudienceProvider audienceProvider) {
 		return new AudienceValidator(audienceProvider);
 	}
 
 	@Bean
 	@ConditionalOnMissingBean(name = "iapJwtDelegatingValidator")
 	public DelegatingOAuth2TokenValidator<Jwt> iapJwtDelegatingValidator(IapAuthenticationProperties properties,
-			ObjectProvider<AudienceValidator> audienceVerifier) {
+			AudienceValidator audienceValidator) {
 
 		List<OAuth2TokenValidator<Jwt>> validators = new ArrayList<>();
 		validators.add(new JwtTimestampValidator());
 		validators.add(new JwtIssuerValidator(properties.getIssuer()));
+		validators.add(audienceValidator);
+
+		if (LOGGER.isInfoEnabled()) {
+			LOGGER.info("IAP JWT Audience validation enabled for audience: " + audienceValidator.getAudience());
```

Comment:
Since audience validation is always on. Something like:
```suggestion
			LOGGER.info("Audience configured for IAP JWT validation: " + audienceValidator.getAudience());
```